nipype/algorithms/misc.py

```diff
@@ -1380,6 +1380,51 @@ def merge_rois(in_files, in_idxs, in_ref,
     return out_file
 
 
+class CalculateMedianInputSpec(BaseInterfaceInputSpec):
+    in_file = InputMultiPath(File(exists=True, mandatory=True,
+      desc="One or more realigned Nifti 4D timeseries"))
+    median_file = traits.Str('median.nii.gz', usedefault=True,
+      desc="Filename to store median image")
```

Comment:
perhaps add a boolean flag to indicate whether the median should be computed individually or across files. and generate the appropriate outputs based on the flag.